src/realtime/synchronizing-custom-data.md

```diff
@@ -16,14 +16,14 @@ For this example, I’m going to make a realtime component that synchronizes the
 
 ### Creating a realtime model
 
-First, we’re going to start by writing the model. Start by removing `MonoBehaviour` from the class definition. Models shouldn't be a subclass of any other class. Next, add a private field to hold our color value. One quick note: variables need to be private and they need to start with an underscore. Once you've added your property, your class should look something like this:
+First, we’re going to start by writing the model. Start by removing `MonoBehaviour` from the class definition. Models shouldn't be a subclass of any other class. Models also need to be `partial` classes so that Normcore can supplement the class with the networking code. Next, add a private field to hold our color value. One quick note: variables need to be private and they need to start with an underscore. Once you've added your property, your class should look something like this:
```

Comment:
Is it possible to make the source gen stuff add partial for us like the old model compiler used to do? I really like that new developers don't have to worry about this / know what it is.

Hmm, I guess that would require us to bring back the button to run the old model compiler. Is there any way we can automate it? or use a hook somewhere to insert partial before the source files are compiled?

Reply:
Unfortunately `partial` has to be added by the author. Source generators can only create new source code; the process can't modify existing code.

I don't think this is an onerous requirement though. I think anyone who would get tripped up on `partial` is already copy-pasting our sample code, and that will still work. For newcomers to C#, `partial` and attributes feel about the same in terms of features-they-might-not-know-about, and we have no qualms about requiring the attribute.

I do have a task to improve the error logging around non-partial models: https://github.com/NormalVR/Normcore/issues/430 Right now you'll just get the standard compiler error for duplicate class definitions but instead we can skip source gen and log a more descriptive error.

Reply:
I agree with this. I think then I'd want to make sure this PR adds `partial` to any areas in the docs where it's missing it, and in this guide, I think we should include a copy of the error that you'll get if it's missing so this section comes up if you search for the error.